### PR TITLE
MINOR: fix incorrect exception message

### PR DESCRIPTION
**Pull request by stepio**
While playing with client got the next exception:

``` java
java.lang.IllegalArgumentException: Invalid partition given with record: 1 is not in the range [0...1].
```

It's obviously incorrect, so I've fixed it.

### CODE REVIEW

**Comment by stepio** [comment]
I see no errors in test results, but build is failed. Any issue on my side?

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 1]
```diff
@@ -711,13 +711,11 @@ private int partition(ProducerRecord<K, V> record, byte[] serializedKey , byte[]
         Integer partition = record.partition();
         if (partition != null) {
             List<PartitionInfo> partitions = cluster.partitionsForTopic(record.topic());
-            int numPartitions = partitions.size();
+            int numPartitions = partitions.size() - 1;
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: This change made the numPartitions variable name inconsistent with semantic meaning. How about changing line 716 to `if (partition < 0 || partition > numPartitions - 1)` instead?

Comment (stepio) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: The actual purpose of this change was to use the same value not only in the check, but also in IllegalArgumentException instantiation.

So two options are possible:
1. Either use "numPartitions - 1" in both places.
2. Or rename the variable to "lastPartition".

I'd prefer the option 2.

Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: Cool. Sounds good to me.

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk. Thanks @stepio !

**Pull request closed without merging**